src/main/java/org/alfresco/repo/jscript/RhinoScriptProcessor.java

```diff
@@ -593,6 +598,24 @@ public Scriptable wrapAsJavaObject(Context cx, Scriptable scope, Object javaObje
             }
             return super.wrapAsJavaObject(cx, scope, javaObject, staticType);
         }
+    }
+    
+    
+    /**
+     * A {@link WrapFactory} that ensures {@link org.mozilla.javascript.NativeJavaObject} instances are of the
+     * {@link SandboxNativeJavaObject} variety.
+     */
+    private static class SandboxWrapFactory extends WrapFactory
```

Comment:
1. Why do we have a special RhinoWrapFactory?
2. Shouldn't then our SandboxWrapFactory extend this RhinoWrapFactory